include/numerics/sparse_matrix.h

```diff
@@ -171,6 +171,13 @@ class SparseMatrix : public ReferenceCountedObject<SparseMatrix<T> >,
   virtual void close () = 0;
 
   /**
+   *  For PETSc matrix , this function is similar to close but without shrinking memory.
+   *  This is useful when we want to switch between ADD_VALUES and INSERT_VALUES.
+   *  close should be called before using the matrix.
+   */
+  virtual void flush () = 0;
```

Comment:
Let's make the default implementation call `close()` instead of being pure virtual. Then you can drop the changes from the Trilinos and Eigen specializations. Also, this PR misses the Laspack implementation (because it's not compiled) but adding a default base class behavior fixes this as well.